back-end/src/reservations/reservations.controller.js

```diff
@@ -138,6 +138,39 @@ async function reservationExists(req, res, next) {
   });
 }
 
+function reservationStatusIsBooked(req, res, next) {
+  const { data: { status = "booked" } = {} } = req.body;
+  if (status === "booked") {
+    return next();
+  }
+  next({
+    status: 400,
+    message: `Reservation status is "${status}". Must be set to "booked" in order to seat`,
+  });
+}
+
+function statusIsNotUnknown(req, res, next) {
+  const { data: { status } = {} } = req.body;
+  if (status === "booked" || status === "seated" || status === "finished") {
+    return next();
+  }
+  next({
+    status: 400,
+    message: `Reservation status is unknown. Valid statuses are "booked", "seated", and "finished".`,
+  });
+}
+
+function statusIsNotFinished(req, res, next) {
+  const { status } = res.locals.reservation;
+  if (status === "finished") {
```

Comment:
nitpick that doesnt need to be done for this proj: i would probably store these statuses in some sort of constant or as strictly typed as you can get without using typescript. its flakey to reuse just strings and would be easier to store these in a keyed constant at a central location that you can use application wide